host/libhackrf/src/hackrf.h

```diff
@@ -442,74 +1204,252 @@ extern ADDAPI int ADDCALL hackrf_init_sweep(
 	const uint32_t offset,
 	const enum sweep_style style);
 
-/* Operacake functions */
+/**
+ * Query connected operacake boards
+ * 
+ * Returns a @ref HACKRF_OPERACAKE_MAX_BOARDS size array of addresses, with @ref HACKRF_OPERACAKE_ADDRESS_INVALID as a placeholder
+ * 
+ * Requires USB API version 0x1002 or higher!
+ * @param[in] device device to query
+ * @param[out] boards list of boards
+ * @return @ref HACKRF_SUCCESS on success or @ref hackrf_error variant
+ */
 extern ADDAPI int ADDCALL hackrf_get_operacake_boards(
 	hackrf_device* device,
 	uint8_t* boards);
 
+// TODO: why arent't all functions use adresses?
```

Comment:
How does that work? I know we have a limit `HACKRF_OPERACAKE_MAX_FREQ_RANGES` (set to 8), so how does specifying more than that work? Will the ranges be distributed across the boards (in groups of 8)? In what order then?

Reply:
Sorry, I'd misunderstood - the [docs clarify that](https://hackrf.readthedocs.io/en/latest/opera_cake.html):

> Although multiple Opera Cakes on a single HackRF One may be set to frequency mode at the same time, they share a single switching plan. This can be useful, for example, for a filter bank consisting of eight filters.

and:

> Although multiple Opera Cakes on a single HackRF One may be set to time mode at the same time, they share a single switching plan.

So the reason those functions don't take an address parameter is that the effects apply to all addresses.